src/helpers/helpers.dom.js

```diff
@@ -162,8 +162,8 @@ export function retinaScale(chart, forceRatio, forceStyle) {
   // making the chart visually bigger, so let's enforce it to the "correct" values.
   // See https://github.com/chartjs/Chart.js/issues/3575
   if (canvas.style && (forceStyle || (!canvas.style.height && !canvas.style.width))) {
-    canvas.style.height = height + 'px';
-    canvas.style.width = width + 'px';
+    canvas.style.height = (canvas.height / pixelRatio) + 'px';
```

Comment:
I don't think this is the correct fix as it will break retina screens. In the case of a retina display, `pixelRatio` will be a number greater than 1. If we need to round, it feels like it should be `Math.floor(canvas.height)` but I suspect there will be cases where this introduces oscillation during resizing. We saw that a lot in previous implementations of the chart resizing because when the height is reduced from say 300.8 to 300, the need for a scrollbar may disappear triggering another resize which increases the width slightly triggering the scrollbar to re-appear which changes the width leading to an infinite loop.

Reply:
When `canvas.height` is assigned, the value is already implicitly rounded. So no additional flooring should be required.

Regarding potential oscillations: I understand your concerns, however, I do not see a good solution. The following naive ideas come to my mind:
- Wrap the canvas in yet another `div` and apply the unrounded dimensions to this `div` while rounding the dimensions for the `canvas`.
- Add the discrepancy between unrounded and rounded values as padding to the canvas.
- Trade the blurring problem for the potential oscillation problem while assuming the blurring does occur far more often than the oscillation would occur.

Reply:
Interesting. So the width/height are multiplied with pixel ratio and implicitly converted to integers. Using that integer value and dividing it with the ratio should actually result in more accurate result even for retina displays.

Not quite sure if this case has been tested previously.

Reply:
Ah, ok that makes more sense. I don't know how we'd get test coverage on this, but the explanation makes sense. If we could adjust to add a comment here explaining why we're doing this, I'm ok with the change